prost-build/src/context.rs

```diff
@@ -0,0 +1,285 @@
+            .unwrap()
+            .field
+            .iter()
+            .all(|field| self.can_message_field_derive_copy(fq_message_name, field))
+    }
+
+    /// Returns `true` if the type of this message field allows deriving the Copy trait.
+    pub fn can_message_field_derive_copy(
+        &self,
+        fq_message_name: &str,
+        field: &FieldDescriptorProto,
+    ) -> bool {
+        self.can_field_derive_copy_impl(fq_message_name, None, field)
+    }
+
+    /// Returns `true` if the type of this field in a oneof allows deriving the Copy trait.
+    pub fn can_oneof_field_derive_copy(
+        &self,
+        fq_message_name: &str,
+        oneof_name: &str,
+        field: &FieldDescriptorProto,
+    ) -> bool {
+        self.can_field_derive_copy_impl(fq_message_name, Some(oneof_name), field)
+    }
+
+    fn can_field_derive_copy_impl(
+        &self,
+        fq_message_name: &str,
+        oneof: Option<&str>,
+        field: &FieldDescriptorProto,
+    ) -> bool {
+        assert_eq!(".", &fq_message_name[..1]);
+
+        // repeated field cannot derive Copy
+        if field.label() == Label::Repeated {
+            false
+        } else if field.r#type() == Type::Message {
+            // nested and boxed messages cannot derive Copy
+            if self
+                .message_graph
+                .is_nested(field.type_name(), fq_message_name)
+            {
+                return false;
+            }
+            let config_path = match oneof {
+                None => Cow::Borrowed(fq_message_name),
+                Some(oneof_name) => Cow::Owned(format!("{fq_message_name}.{oneof_name}")),
+            };
+            if self
+                .config
+                .boxed
+                .get_first_field(&config_path, field.name())
+                .is_some()
+            {
+                false
+            } else {
+                self.can_message_derive_copy(field.type_name())
+            }
+        } else {
+            matches!(
+                field.r#type(),
+                Type::Float
+                    | Type::Double
+                    | Type::Int32
+                    | Type::Int64
+                    | Type::Uint32
+                    | Type::Uint64
+                    | Type::Sint32
+                    | Type::Sint64
+                    | Type::Fixed32
+                    | Type::Fixed64
+                    | Type::Sfixed32
+                    | Type::Sfixed64
+                    | Type::Bool
+                    | Type::Enum
+            )
+        }
+    }
+
+    pub fn should_disable_comments(&self, fq_message_name: &str, field_name: Option<&str>) -> bool {
+        if let Some(field_name) = field_name {
+            self.config
+                .disable_comments
+                .get_first_field(fq_message_name, field_name)
+                .is_some()
+        } else {
+            self.config
+                .disable_comments
+                .get_first(fq_message_name)
```

Comment:
`PathMap::get_first` is subtly different from `PathMap::get` and then calling `.next()`. Is that intentional?

Reply:
No. Would it behave differently in any cases?

Reply:
I have not dug into that. I just see that the code does something different 😉

I suggest using the old code and opening another (smaller) PR for adding tests to prove noting changed.